src/librustc/middle/ty.rs

```diff
@@ -749,6 +749,9 @@ pub struct ctxt<'tcx> {
     /// Maps a trait onto a list of impls of that trait.
     pub trait_impls: RefCell<DefIdMap<Rc<RefCell<Vec<ast::DefId>>>>>,
 
+    /// Maps a trait onto a list of negative impls of that trait.
+    pub trait_negative_impls: RefCell<DefIdMap<Rc<RefCell<Vec<ast::DefId>>>>>,
```

Comment:
from what I can see, there is no reason to keep these in separate lists, we always just want to consider the union of the two vectors, no?